ddl/placement/bundle.go

```diff
@@ -124,6 +125,18 @@ func NewBundleFromConstraintsOptions(options *model.PlacementSettings) (*Bundle,
 		}
 	}
 
+	survivalPreferenceStr := options.SurvivalPreferences
+	if len(survivalPreferenceStr) > 0 {
+		labels := []string{}
+		err := yaml.UnmarshalStrict([]byte(survivalPreferenceStr), &labels)
+		if err != nil {
+			return nil, ErrInvalidConstraintsFormat
+		}
```

Comment:
I would like to see an new error, or possibly a new function for parsing `survivalPreference`.